enclave/src/state.rs

```diff
@@ -162,8 +162,10 @@ pub fn list_shards() -> SgxResult<Vec<ShardIdentifier>> {
 }
 
 //  tests
+#[cfg(feature = "test")]
```

Comment:
Shall we not put that in a `mod test` to prevent pollution with `#[cfg(feature = "test")]`? If we want to avoid changing the public api for this lib, we can still do re-export the test mod's entries with:

```
#[cfg(feature = "test")]
pub use test::*;
```

Reply:
I've done that in some places, yes - but not everywhere, in order to keep this PR 'small'. But I guess I can expand these changes while I'm at it.

Reply:
Now looking at it, this seems to have been the only place where I did not do that refactoring 🙈  so it should absolutely be done here as well

Reply:
haha, glad to see that I am not the only one suffering from the inconsistency-syndrome.

Reply:
not by a long shot - most developers are guilty of that at times 😄